File: src/main/java/frc/robot/commands/climber/ManualFangsControl.java
package frc.robot.commands.climber;

import edu.wpi.first.wpilibj.command.Command;
import frc.robot.Robot;
import frc.robot.RobotMap;
import frc.robot.util.MercMath;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

/**
 * Manual control for the fangs
 */
public class ManualFangsControl extends Command {
    private final Logger LOG = LogManager.getLogger(ManualFangsControl.class);

    public ManualFangsControl() {
        requires(Robot.fangs);
        setName("ManualFangsControl Command");
        LOG.info(getName() + " Constructed");
    }

    /**
     * Articulate the fangs in accordance with the gamepad axis
     */
    @Override
    protected void execute() {
        Robot.fangs.getArticulator().setSpeed(MercMath.applyDeadzone(Robot.oi.getGamepadAxis(RobotMap.GAMEPAD_AXIS.leftY), 0.1));
        LOG.info(getName() + " Executed");
    }

    @Override
    protected boolean isFinished() {
        return false;
    }

    @Override
    protected void end() {
        LOG.info(getName() + " Ended");
    }

    @Override
    protected void interrupted() {
        LOG.info(getName() + " Interrupted");
        this.end();
    }
}

File: src/main/kotlin/ktfrc/robot/commands/climber/ManualFangsControl.kt
package ktfrc.robot.commands.climber

import edu.wpi.first.wpilibj.command.Command
import ktfrc.robot.RobotMap
import ktfrc.robot.Robot
import ktfrc.robot.util.applyDeadzone
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

class ManualFangsControl: Command() {

    init {
        requires(Robot.fangs)
        name = "ManualFangsControl Command"
        LOG?.info("$name Constructed")
    }

    override fun execute() {
        Robot.fangs.articulator.setSpeed(applyDeadzone(Robot.oi.getGamepadAxis(RobotMap.GAMEPAD_AXIS.leftY), 0.1))
        LOG?.info("$name Executed")
    }

    override fun isFinished(): Boolean = false

    override fun end() {
        LOG?.info("$name Ended")
    }

    override fun interrupted() {
        LOG?.info("$name Interrupted")
        this.end()
    }

    companion object {
        val LOG: Logger? = LogManager.getLogger(ManualFangsControl::class.java)
    }
}